pkg/manager/member/tiproxy_member_manager.go

```diff
@@ -0,0 +1,656 @@
+const tiproxyCertVolumeMountPath = "/var/lib/tiproxy-tls"
+
+func labelTiProxy(tc *v1alpha1.TidbCluster) label.Label {
+	instanceName := tc.GetInstanceName()
+	return label.New().Instance(instanceName).TiProxy()
+}
+
+// tiproxyMemberManager implements manager.Manager.
+type tiproxyMemberManager struct {
+	deps      *controller.Dependencies
+	scaler    Scaler
+	upgrader  Upgrader
+	suspender suspender.Suspender
+}
+
+// NewTiProxyMemberManager returns a *tiproxyMemberManager
+func NewTiProxyMemberManager(deps *controller.Dependencies, scaler Scaler, upgrader Upgrader, spder suspender.Suspender) manager.Manager {
+	m := &tiproxyMemberManager{
+		deps:      deps,
+		scaler:    scaler,
+		upgrader:  upgrader,
+		suspender: spder,
+	}
+	return m
+}
+
+// Sync fulfills the manager.Manager interface
+func (m *tiproxyMemberManager) Sync(tc *v1alpha1.TidbCluster) error {
+	if tc.Spec.TiProxy == nil {
+		return nil
+	}
+
+	ns := tc.GetNamespace()
+	tcName := tc.GetName()
+
+	if tc.Spec.Paused {
+		klog.Infof("TidbCluster %s/%s is paused, skip syncing tiproxy service", ns, tcName)
+		return nil
+	}
+
+	// skip sync if tiproxy is suspended
+	component := v1alpha1.TiProxyMemberType
+	needSuspend, err := m.suspender.SuspendComponent(tc, component)
+	if err != nil {
+		return fmt.Errorf("suspend %s failed: %v", component, err)
+	}
+	if needSuspend {
+		klog.Infof("component %s for cluster %s/%s is suspended, skip syncing", component, ns, tcName)
+		return nil
+	}
+
+	// If PD is available, wait it
+	if tc.Spec.PD != nil && !tc.PDIsAvailable() {
+		return controller.RequeueErrorf("TidbCluster: [%s/%s], TiProxy is waiting for PD cluster running", ns, tcName)
+	}
+
+	if err := m.syncProxyService(tc, false); err != nil {
+		return err
+	}
+	if err := m.syncProxyService(tc, true); err != nil {
+		return err
+	}
+
+	return m.syncStatefulSet(tc)
+}
+
+func (m *tiproxyMemberManager) syncConfigMap(tc *v1alpha1.TidbCluster, set *apps.StatefulSet) (*corev1.ConfigMap, error) {
+	PDAddr := fmt.Sprintf("%s:2379", controller.PDMemberName(tc.Name))
+	// TODO: support it
+	if tc.AcrossK8s() {
+		return nil, fmt.Errorf("across k8s is not supported for")
+	}
+	if tc.Heterogeneous() && tc.WithoutLocalPD() {
+		PDAddr = fmt.Sprintf("%s:2379", controller.PDMemberName(tc.Spec.Cluster.Name)) // use pd of reference cluster
+	}
+
+	state := map[string]interface{}{
+		"WorkDir": filepath.Join(tiproxyVolumeMountPath, "work"),
+		"PDAddr":  PDAddr,
+	}
+	if tc.IsTLSClusterEnabled() {
+		state["PeerTLS"] = tiproxyCertVolumeMountPath
+		state["ClusterClientTLS"] = util.ClusterClientTLSPath
+	}
+	if tc.Spec.TiDB != nil && tc.Spec.TiDB.IsTLSClientEnabled() && !tc.SkipTLSWhenConnectTiDB() {
+		state["SQLClientTLS"] = tidbClientCertPath
+	}
+
+	confBuf := &bytes.Buffer{}
+	if err := template.Must(template.New("config").Parse(`
+workdir: {{ .WorkDir }}
+proxy:
+  addr: "0.0.0.0:6000"
+  tcp-keep-alive: true
+  max-connections: 1000
+  pd-addrs: {{ .PDAddr }}
+metrics:
+  prom-cluster: "default"
```

Comment:
Not directly. You can change config in `TidbCluster CR ` but can not pass tiproxy config directly. We have too many integration for tiproxy(it may even be merged with dev tier gateway later), thus we decided to expose less config.

Reply:
> thus we decided to expose less config

so, will some config support be updated online?

Reply:
It already is. https://github.com/pingcap/tidb-operator/blob/24743fbce41ea7f416a2d11d0212853286216566/pkg/manager/member/tiproxy_member_manager.go#L316-L322

Reply:
After supporting clustering, config will be applied to all nodes automatically.

Reply:
> so, will some config support be updated online?

Do you mean dynamic config without updating CRD?